src/hooks/index.js

```diff
@@ -0,0 +1,53 @@
+import axios from 'axios'
+import { useState, useEffect } from 'react'
+
+
+/**
+ * A custom hook that can be used directly or by other hooks to make axios calls
+ * @param {string} url url endpoint to hit
+ * @param {string} method method of request
+ * @param {object} data data to send in post request, for example
+ * @param {object} config config obejct
+ */
+
+export const useAxios = (url, method, data, config) => {
+  const [axiosData, setAxiosData] = useState(null);
+  const [loading, setLoading] = useState(true);
+  useEffect(() => {
+    const fetchData = async function () {
+      try {
+        setLoading(true);
+        const response = await axios[method](url, data, config);
```

Comment:
It could help to add error handling, ie check that the passed in method is an approved one